src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,551 @@
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void initVariable() {
+        formatterAdminTime = new SimpleDateFormat("dd/MM/yy");
+        formatterAdminTime.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
     }
 
     @Override
-    @Test
     public void testExecuteAndPostProcess() {
+        // See each independent test cases
+    }
+
+    @BeforeGroups("typicalLogs")
+    public void removeAndRestoreLogMessage() {
+        gaeSimulation.loginAsAdmin("admin");
+        gaeSimulation.clearLogs();
+
+        Date twoDaysAgo = TimeHelper.getDateOffsetToCurrentTime(-2);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_TWO_DAYS_AGO), twoDaysAgo.getTime());
+        Date yesterday = TimeHelper.getDateOffsetToCurrentTime(-1);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_YESTERDAY), yesterday.getTime());
+        Date today = TimeHelper.getDateOffsetToCurrentTime(0);
+        insertLogMessagesAtTime(logMessages.get(LOG_MESSAGE_INDEX_TODAY), today.getTime());
+    }
+
+    @Test(groups = "typicalLogs")
+    public void filterQuery_invalidQuery_defaultSearchPerformed() {
+        int[][] expected = new int[][]{{0, 1, 3, 4, 5}};
+        String query = "unknown";
+        verifyActionResult(expected, "filterQuery", query);
+        query = "";
+        verifyActionResult(expected, "filterQuery", query);
+        query = "info";
+        verifyActionResult(expected, "filterQuery", query);
+        query = "info:";
+        verifyActionResult(expected, "filterQuery", query);
+        query = "request:servlet3 unknown_connector role:Student";
+        verifyActionResult(expected, "filterQuery", query);
+        query = "unknown:servlet3 | role:Student";
+        verifyActionResult(expected, "filterQuery", query);
+
+        // invalid filterQuery with showing testing data
+        expected = new int[][]{{0, 1, 2, 3, 4, 5, 6}};
+        query = "information:unkown";
+        verifyActionResult(expected, "filterQuery", query, "testdata", "true");
+    }
+
+    @Test(groups = "typicalLogs")
+    public void urlParams_testDataAndAll_showTestDataAndExcludedUriAccordingly() {
+        // no test data, no excluded URI, default search
+        int[][] expected = new int[][]{{0, 1, 3, 4, 5}};
+        verifyActionResult(expected);
+
+        // show test data, no excluded URI
+        expected = new int[][]{{0, 1, 2, 3, 4, 5, 6}};
+        verifyActionResult(expected, "testdata", "true");
+
+        // not test data, show excluded URI
+        expected = new int[][]{{0, 1, 3, 4, 5, 7, 8, 9}};
+        verifyActionResult(expected, "all", "true");
+
+        // show everything
+        expected = new int[][]{{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}};
+        verifyActionResult(expected, "testdata", "true", "all", "true");
+    }
+
+    @Test(groups = "typicalLogs")
+    public void filterQuery_validQuery_querySuccessful() {
```

Comment:
Unless you have another case `filterQuery_validQuery_queryFails()` i.e. variations of the 3rd component, you can omit the 3rd component altogether. e.g. `filterQuery_validQuery()`